filters/openpolicyagent/authorizewithregopolicy/authorizewithregopolicy.go

```diff
@@ -0,0 +1,163 @@
+package authorizewithregopolicy
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/yaml.v2"
+
+	"github.com/zalando/skipper/filters/openpolicyagent"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/envoy"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/util"
+)
+
+type Spec struct {
+	factory    openpolicyagent.OpenPolicyAgentFactory
+	configOpts []func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error
+}
+
+func NewAuthorizeWithRegoPolicySpec(factory openpolicyagent.OpenPolicyAgentFactory, opts ...func(*openpolicyagent.OpenPolicyAgentInstanceConfig) error) Spec {
+	return Spec{
+		factory:    factory,
+		configOpts: opts,
+	}
+}
+
+func (s Spec) Name() string {
+	return "authorizeWithRegoPolicy"
+}
+
+const (
+	paramBundleName int = iota
+	paramEnvoyContextExtensions
+)
+
+func (s Spec) CreateFilter(config []interface{}) (filters.Filter, error) {
+	var err error
+
+	sargs, err := util.GetStrings(config)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sargs) < 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if len(sargs) > 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	bundleName := sargs[paramBundleName]
```

Comment:
I would say keep it simple without const and use plain `sargs[0]` and `sargs[1]`.